tests/loops/test_loops.py

```diff
@@ -334,13 +332,151 @@ def val_dataloader(self):
     assert trainer.fit_loop.epoch_loop.val_loop.epoch_loop.batch_progress.state_dict() == expected
 
 
+@RunIf(min_torch="1.7.0")
+@mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"})
+@pytest.mark.parametrize("stop_epoch", (0, 1))
+@pytest.mark.parametrize("stop_batch", (0, 1, 2))
+@pytest.mark.parametrize("n_optimizers,stop_optimizer", [(2, 0), (2, 1), (3, 2)])
+def test_loop_restart_progress_multiple_optimizers(tmpdir, n_optimizers, stop_optimizer, stop_epoch, stop_batch):
```

Comment:
recommendations where such a test should live? here? in `tests/utilities/test_auto_restart.py`

Reply:
also, the test runs for about 3-4 seconds on my CPU. not sure how to cut it down. if we want to compare the actual trained weights, we need to run a training. 
perhaps cut down on the parametrizations? (though it is important to cover the boundary cases)

Reply:
`tests/loops/optimization/test_optimizer_loop.py`?

Reply:
Can you skip validation by setting limit_val_batches=0 and logger=None and checkpointing ?

Reply:
done, both